test/unit/muxed_account_test.js

```diff
@@ -0,0 +1,59 @@
+describe('muxed account abstraction works', function() {
+  const PUBKEY = 'GA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJVSGZ';
+  const MPUBKEY_ZERO =
+    'MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAACJUQ';
+  const MPUBKEY_ID =
+    'MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAABUTGI4';
+  const MPUBKEY_ID2 =
+    'MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAABQHF6OU';
+
+  const performBasicInvariantChecks = function(mux, id, expectedM) {
+    expect(mux.accountId()).to.equal(PUBKEY);
+    expect(mux.address()).to.equal(expectedM);
+    expect(mux.id()).to.equal(id);
+
+    const muxXdr = mux.asXDRObject();
+    expect(muxXdr.switch()).to.equal(
+      StellarBase.xdr.CryptoKeyType.keyTypeMuxedEd25519()
+    );
+
+    const innerMux = muxXdr.med25519();
+    expect(innerMux.ed25519()).to.eql(
+      StellarBase.StrKey.decodeEd25519PublicKey(PUBKEY)
+    );
+    expect(innerMux.id()).to.eql(StellarBase.xdr.Uint64.fromString(id));
+
+    return muxXdr;
+  };
+
+  it('can construct from a G address', function() {
+    const mux = new StellarBase.MuxedAccount(PUBKEY);
+    performBasicInvariantChecks(mux, '0', MPUBKEY_ZERO);
+  });
```

Comment:
`G` aren't really muxed accounts though, right? They're just a normal account?

Reply:
Right, so this is to create a muxed account (w/ an ID of zero) from a G account.

Reply:
(in other words, this abstraction is opt-in by using it)